unittests/test_pipeline.py

```diff
@@ -457,6 +457,37 @@ def postfoo(self):
         assert '_RFM_PRE_SETUP' in os.environ
         assert '_RFM_POST_SETUP' in os.environ
 
+    def test_setup_hooks_in_compile_only_test(self):
+        @fixtures.custom_prefix('unittests/resources/checks')
+        class MyTest(rfm.CompileOnlyRegressionTest):
+            def __init__(self):
+                self.name = 'hellocheck_compile'
+                self.descr = 'C Hello World test'
+
+                # All available systems are supported
+                self.valid_systems = ['*']
+                self.valid_prog_environs = ['*']
+                self.sourcepath = 'hello.c'
+                self.tags = {'foo', 'bar'}
+                self.name = type(self).__name__
+                self.executable = os.path.join('.', self.name)
```

Comment:
You should not care about this.

Reply:
I think that you can go without defining the executable.

Reply:
I have to define the executable name because otherwise the build script has this line `gcc hello.c -o ./TestHooks.test_setup_hooks_in_compile_only_test.<locals>.MyTest` and fails with the error `.../unittests/tmp52k816ku/stage/generic/login/builtin-gcc/hellocheck_compile/rfm_hellocheck_compile_build.sh: line 12: locals: No such file or directory`. Do you have any suggestion how I could avoid it?

Reply:
Ok, not a problem then to leave it as of now.